3_import/bulk_import.py

```
@@ -248,8 +248,14 @@ def process_step(step, client, outputs, outputs_file_name):
                 return False
             data = data[0]
         elif action['method'] == 'post':
-            data = client.post(full_url, json=body)
-
+          if step['action'] == 'create-farm':
+             name = body['name'].replace(" ", "%20")
+             data = client.list(full_url + 'name=' + name)
+             if len(data) != 1:
```

I assume that this is to prevent recreating the farm if it already exists, but that situation should never happen under normal circumstances. Did Ryan encounter it with customers? If so do you know what led to that? I'm wondering if it would be better to just ignore the error raised by Scalr when we try to create an object that already exists - this way it would work for farm roles too.
Generally speaking I think it would be better not to add anything that depend on the action in this function to keep it simple. Maybe a generic solution would be to add a list of ignored error codes to the step - this way we can customize the behavior during the plan creation.

---

@AloysAugustin a couple new requirements:
"In general, the bulk import should be able to re-run on the same import file generating errors because of subsequent runs. If a farm is already created, re-run should just skip it. The same principal should apply to farm role, project, and host."

"Adding a few more context... we plan to use the bulk import to perpetually import any new hosts launched outside of Scalr. This import will be running as a daemon process. In summary the daemon process will 1) scan aws for newly launched host (say in the last 8 hours and without scalr-meta tag) 2) map the host records to produce scalr bulk import interface file 3) bulk import will take this interface file, determine what need to be created, such as farm, farm role, and project, and if these are already exist, skip the setup step, run the host import if the host is not already imported."

Taking this step by step and starting with farms, farm roles, and servers. Still have a lot of discussion around the full scope.

---

Thanks for the context, that's helpful. That's quite different from the initial requirements I built this for but we can probably make it work ;-) . Knowing this I think ignoring the conflict error when we try to create farms and farm roles that already exist is the easiest way to achieve what they want.